src/SecretsApi/SecretsApi.php

```diff
@@ -144,15 +158,17 @@ public function setSecret(
             $scopes = array_map('trim', explode(',', $scopes));
             $body['scopes'] = $scopes;
         }
-        $options = [
-            'headers' => [
-                'Accept' => 'application/json',
-                'Authorization' => $this->request()->session()->get('session'),
-            ],
-            'json' => $body,
-            'method' => 'POST',
-            'debug' => $debug,
-        ];
+
+        if($env_name) {
+            $url = sprintf('%s/sites/%s/secrets/%s', $this->getBaseURI(), $site_id, $name);
+            $body['env'] = $env_name;
+            $options['method'] = 'PATCH';
+            unset($body['name']);
```

Comment:
I'm not super fan of setting and then unsetting stuff but I guess we can fix later?

Reply:
Not sure if there's a way to avoid this without using an `else` statement 😳

And greg said `if(env)`, `if(!env)` was even worse than `else`...